src/main/java/chatbox/ChatMessage.java

```diff
@@ -28,7 +30,9 @@
 
     private final Date creationTime = new Date();
 
-    @NotBlank
+    @NotBlank(message = "Message must have at least " + SmokefreeConstants.ChatBox.MINMUM_MESSAGE_LENGTH + " characters")
+    @Pattern(regexp = "^[A-Za-z0-9\\n!@&(),.?\": ]+$", message = "Please enter only allowed special charaxters: @&(),.?\": ")
```

Comment:
need to check what characters we should allow

Reply:
drop the character whitelist. We will rely on the front-end to escape.
Please do not yet merge this pr